doc/firewalld_services.md

```diff
@@ -0,0 +1,83 @@
+```
+
+In most of the cases the only requirement will be to define the service in `firewalld` and probabily it will be already
+provided by the `firewalld` package. In summary the changes in code should look like:
+
+```ruby
+
+# Require the new firewalld library and drop any import of SuSEFirewall2
+require 'y2firewall/firewalld'
+
+  # In your Module.Read method replace SuSEFirewalld.Read by firewalld.read
+    firewalld.read
+  # # In your Module.Write method replace SuSEFirewalld.Write by firewalld.read
+    firewalld.write
+  
+  private
+  
+    # Of course also define the firewalld convenience method to obtain the firewalld singlenton instance
+    def firewalld
+      Y2Firewall::Firewalld.instance
+    end
+```
+
+### Modify service ports definition
+
+The services definition that cames with firewalld can be modified. By default all the services that came with firewalld are placed in `/usr/lib/firewalld/services` although if a service is modified then it is placed in `/etc/firewalld/services` allowing the admin to go bak to the original definition is needed.
+
+SuSEFirewallServices was used in the past for modifying the services definition, it has been also dropped and the services configuration have to be done through the new class `Y2Firewall::Firewalld::Service`. 
+
+To modify the ports definition, the class method `modify_ports` has been provided making the call as similar to the old one (`SetNeededPortsAndProtocols`) as possible.
+
+```ruby
+      
+      # SuSEFirewallServices.SetNeededPortsAndProtocols("service:cluster", { "tcp_ports" => tcp_ports })
+      begin
+        Y2Firewall::Firewalld::Service.modify_ports(name: "cluster", tcp_ports: tcp_ports)
+      rescue Y2Firewall::Firewalld::Service::NotFound
+        y2error("Firewalld 'cluster' service is not available.")
+      end
+```
+
+To get the list of service ports we have to ask the service object itself.
+
+```ruby
+      # SuSEFirewallServices.GetNeededTCPPorts("service:cluster")
+      begin
+        fwd_cluster = firewalld.find_service("cluster")
+        tcp_ports = fwd_cluster.tcp_ports
+      rescue Y2Firewall::Firewalld::Service::NotFound
+        tcp_ports = []
+      end
+
+```
+
+## Important Note:
+
+If a `firewalld service` is not defined, then the `CWMFirewallInterfaces` widget will show a summary of not available services, and will suggest to deploy them to be able to configure the firewall.
+
+![notfoundservice](https://user-images.githubusercontent.com/7056681/35266550-87617114-001b-11e8-9135-512925964891.png)
+
+New `services` can be created through the **API** or by a custom xml file although the preferred way is to define them by **RPM**. Please refer to this [link](https://en.opensuse.org/Firewalld/RPM_Packaging) for further information.
+
+In this [PR](https://github.com/yast/yast-cluster/pull/34) a complete example with all the changes needed for replacing `SuSEFirewall2` by `firewalld` is shown.
```

Comment:
What about something like "Find a complete example including all the needed changes in this yast2-cluster PR".